src/adk/managers/auth_manager.py

```diff
@@ -82,14 +85,19 @@ def __read_token(self, host: str) -> Optional[str]:
         """
         return self.__get_item_from_host(host, "token")
 
-    def __store_token(self, host: str, token: str, email: Optional[str], password: Optional[str]) -> None:
-        """Save the token for a host to a file. Currently allowing one login at a time.
+    def __store_token(self, host: str, token: str, email: Optional[str], password: Optional[str],
+                      use_username: Optional[bool]) -> None:
+        """Save the token for a host to a file. Currently, allowing one login at a time.
 
         :param host: the Quantum Network host for which the token is saved.
         :param token: the Quantum Network token to save.
+        :param email: the email to save.
+        :param password: the password to save.
+        :param use_username: the flag to use email or username to save.
         """
         accounts = {}
-        accounts[host] = {"token": token, "email": email, "password": password}
+        accounts[host] = {"token": token, "email": email, "password": password,
+                          "use_username": None if use_username is None else '1' if use_username else '0'}
```

Comment:
Why '1' or '0' rather than just the boolean value itself?

Reply:
I guess I wanted to have string in this configuration instead of a boolean value to keep the type equal. This is a good point. I adjusted to boolean. Some extra changes because of mypy wanted to have clear if the return type is str, bool or None.